core/src/main/java/com/linecorp/armeria/server/RedirectService.java

```diff
@@ -0,0 +1,187 @@
+    }
+
+    /**
+     * Creates a new instance that redirects to the location returned by {@code locationFunction}.
+     *
+     * @param locationFunction a {@link Function} that takes a {@link ServiceRequestContext}
+     *                         and returns a new location.
+     */
+    public RedirectService(Function<? super ServiceRequestContext, String> locationFunction) {
+        this(HttpStatus.TEMPORARY_REDIRECT, locationFunction);
+    }
+
+    /**
+     * Creates a new instance that redirects to the location constructed with the specified
+     * {@code locationPattern}.
+     *
+     * @param redirectStatus the {@link HttpStatus} that the {@link Service} will return.
+     * @param locationPattern the location pattern that is used to generate a redirect location.
+     *
+     * @throws IllegalArgumentException if the specified {@code locationPattern} is unsupported or invalid
+     */
+    public RedirectService(HttpStatus redirectStatus, String locationPattern) {
+        this(redirectStatus, toLocationFunction(locationPattern));
+
+        final Matcher m = PATTERN_PARAMS_START.matcher(locationPattern);
+        if (m.find()) {
+            paramNames = PathMapping.of(locationPattern.substring(m.start())).paramNames();
+        }
+    }
+
+    /**
+     * Creates a new instance that redirects to the location returned by {@code locationFunction}.
+     *
+     * @param redirectStatus the {@link HttpStatus} that the {@link Service} will return.
+     * @param locationFunction a {@link Function} that takes a {@link ServiceRequestContext}
+     *                         and returns a new location.
+     */
+    public RedirectService(HttpStatus redirectStatus,
+                           Function<? super ServiceRequestContext, String> locationFunction) {
+        requireNonNull(redirectStatus, "redirectStatus");
+        requireNonNull(locationFunction, "locationFunction");
+        if (redirectStatus.compareTo(HttpStatus.MULTIPLE_CHOICES) < 0 ||
+            redirectStatus.compareTo(HttpStatus.TEMPORARY_REDIRECT) > 0) {
+            throw new IllegalArgumentException("redirectStatus: " + redirectStatus + " (expected: 300 .. 307)");
+        }
+        httpStatus = redirectStatus;
+        this.locationFunction = locationFunction;
+    }
+
+    /**
+     * NB: For now we redirect all methods.
+     */
+    @Override
+    public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req)
+            throws Exception {
+        return HttpResponse.of(HttpHeaders.of(httpStatus)
+                .set(HttpHeaderNames.LOCATION, locationFunction.apply(ctx)));
+    }
+
+    @Override
+    public void serviceAdded(ServiceConfig cfg) throws Exception {
+        String originPathPattern = cfg.pathMapping().exactPath().orElse(cfg.pathMapping().toString());
+        if (!isDefaultUriPattern(originPathPattern)) {
+            throw new IllegalArgumentException("pathMapping: " + originPathPattern +
```

Comment:
Actually, we don't really need this check at all. For example, a user can use regex-based mapping and still should be allowed to use this service.

Reply:
Hmm youre right, so in the case of 
`glob:/old/*/*` -> `/new/:var1/:var2` we will just have an exception of non matching params
`glob:/old/*/*` -> `/new/:0/:1` it will just map ok